core/kazoo_schemas/src/kz_json_schema_extensions.erl

```diff
@@ -115,3 +116,44 @@ validate_attachment_oauth_doc_id(Value, State) ->
             lager:debug("~s", [ErrorMsg]),
             jesse_error:handle_data_invalid('external_error', ErrorMsg, State)
     end.
+
+maybe_check_params_stability_level(Value, Schema, State) ->
+    case {kz_json:get_value(<<"properties">>, Schema),
+          kz_json:get_value(<<"id">>, Schema),
+          kz_json:is_json_object(Value)}of
+        {'undefined', _, _} ->
+            State; %% Skipping schema with out `properties' property
+        {Properties, Id, true} when Id /= <<"system_config.crossbar">> ->
+            check_params_stability_level(Properties, Value, State);
+        _ ->
+            State
+    end.
+
+check_params_stability_level(Properties, Value, State) ->
+    SystemSL = kapps_config:get_binary(<<"crossbar">>, <<"stability_level">>, <<"stable">>),
+    SystemSLInt = stability_level_to_int(SystemSL),
+    Fun = fun(K, V, Acc) -> check_param_stability_level(Properties, SystemSLInt, K, V, Acc) end,
+    case kz_json:foldl(Fun, [], Value) of
+        [] ->
+            State; %% Not invalid keys found
+        InvalidKeys ->
+            InvalidKeysBin = binary:list_to_bin(lists:join(<<", ">>, InvalidKeys)),
+            ErrorMsg = <<"Disallowed properties -they've lower stability level than"
+                         " system's stability level" " (", SystemSL/binary, ")-: ",
+                         InvalidKeysBin/binary>>,
+            lager:debug("~s", [ErrorMsg]),
+            jesse_error:handle_data_invalid('external_error', ErrorMsg, State)
+    end.
```

Comment:
this seems wrong.
you should check the stability level for the value being checked, not for its children